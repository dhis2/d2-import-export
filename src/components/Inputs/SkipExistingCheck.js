import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_SKIP = { value: 'true', label: i18n.t('Skip check (fast)') }
export const OPTION_CHECK = {
    value: 'false',
    label: i18n.t('Check (safe, recommended)'),
}
export const SKIP_EXISTING_CHECK_KEY = 'skipExistingCheck'
export const SKIP_EXISTING_CHECK_DEFAULT_VALUE = OPTION_CHECK.value

export const SkipExistingCheck = () => (
    <RadioGroup
        name={SKIP_EXISTING_CHECK_KEY}
        label={i18n.t('Skip existing check')}
        options={[OPTION_SKIP, OPTION_CHECK]}
        dataTest="input-skip-existing-check"
    />
)
