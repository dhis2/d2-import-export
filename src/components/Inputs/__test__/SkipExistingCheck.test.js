import React from 'react'
import renderer from 'react-test-renderer'
import { SkipExistingCheck } from '../SkipExistingCheck'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - SkipExistingCheck', () => {
    it('should render correctly', () => {
        const file = renderer.create(<SkipExistingCheck />).toJSON()

        expect(file).toMatchSnapshot()
    })
})
