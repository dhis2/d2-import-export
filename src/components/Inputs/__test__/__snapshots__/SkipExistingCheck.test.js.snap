// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - SkipExistingCheck should render correctly 1`] = `
<RadioGroup
  dataTest="input-skip-existing-check"
  label="Skip existing check"
  name="skipExistingCheck"
  options={
    Array [
      Object {
        "label": "Skip check (fast)",
        "value": "true",
      },
      Object {
        "label": "Check (safe, recommended)",
        "value": "false",
      },
    ]
  }
/>
`;
